File: leetcode/editor/cn/2269_找到一个数字的 K 美丽值_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        ans = self.solution.divisorSubstrings(430043, 2)
        self.assertEqual(2, ans)


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        ans = 0
        # TODO num to string
        # TODO string to num
        # TODO string as slice
        num_str: str = str(num)
        for idx in range(len(num_str) - k + 1):
            val: int = int(num_str[idx:idx + k])
            if val == 0:
                continue
            if num % val == 0:
                ans += 1
        return ans


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()

File: leetcode/editor/cn/2269_找到一个数字的 K 美丽值_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"strconv"
	"testing"
)

func Test_givenNormal_when2269_thenSuccess(t *testing.T) {
	ans := divisorSubstrings(430043, 2)
	assert.Equal(t, 2, ans, "shouldEqual")
}

func Test_givenZero_when2269_thenSuccess(t *testing.T) {
	val := 0
	// error division by zero
	// val = 100 % 0
	assert.Equal(t, 0, val, "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
func divisorSubstrings(num int, k int) int {
	// num to string
	// string to num
	ans := 0
	numStr := strconv.FormatInt(int64(num), 10)
	for i := 0; i <= len(numStr)-k; i++ {
		val, _ := strconv.Atoi(numStr[i : i+k])
		if val == 0 {
			continue
		}
		if num%val == 0 {
			ans += 1
		}
	}
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)
